packages/components/controlled-input/src/ControlledInput.styles.ts

```diff
@@ -0,0 +1,119 @@
+import { css } from 'emotion';
+import tokens from '@contentful/f36-tokens';
+
+export const styles = {
+  container: css({
+    position: 'relative',
+    textOverflow: 'ellipsis',
+    lineHeight: tokens.lineHeightL,
+    margin: '0',
+    width: tokens.spacingM,
+    height: tokens.spacingM,
+    minHeight: tokens.spacingM,
+    overflow: 'visible',
+  }),
+  input: css({
+    display: 'inline-block',
+    fontSize: tokens.fontSizeM,
+    opacity: '0',
+    position: 'absolute',
+    margin: '0',
+    width: tokens.spacingM,
+    height: tokens.spacingM,
+    zIndex: 1,
+    cursor: 'pointer',
+    '&:focus': {
+      '& + label': {
```

Comment:
Nit: this can just be `'&:focus + label': {...}`

Reply:
Thanks for pointing it out. That will definitely save me a few lines in this file. I believe that mostly, this is better. But IMO, it might also decrease readability if it's easier to recognize the same selector block in various contexts.

e,g,
```
'& + label': {
        background: blue;
 },
'&:disabled' {
  '& + label': {
        background: red;
 },
},
'&:checked' {
  '& + label': {
        background: green;
 },
}
```
So in that case, I always have the same selector string '& + label' but easily grasp the different context it is used in (disabled, checked, normal).

If we always prefer merging those selectors, we should consider adding a linting rule. So this won't trigger further such feedback loops as this one :)